ls1intum/Artemis
# Feature/ldap

### Checklist

### Motivation and Context
For the grade export we need the user's registration number if available

### Description
The registration number of all existing users will be loaded once from a given LDAP (configured in yml). In addition, during login, the registration number of TUM students will be loaded once, if it was not loaded before.
The LDAP service is decoupled into a Profile and activating it is optional

### Steps for Testing
1. Activate the LDAP service in the application-artemis.yml file by providing the correct values
2. Log in to ArTEMiS (with a not known TUM account)
3. Navigate to Course Administration
4. Open Scores of one course with TUM students.
5. Export the CSV and check that the student registration number is included
